custom/lib/User.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+namespace Passage\Client;
+
+use OpenAPI\Client\Api\TokensApi;
+use OpenAPI\Client\Configuration;
+use OpenAPI\Client\Api\UsersApi;
+use OpenAPI\Client\Api\UserDevicesApi;
+use OpenAPI\Client\Model\UserInfo as PassageUser;
+use OpenAPI\Client\Model\UpdateUserRequest as UpdateUserArgs;
+use OpenAPI\Client\Model\CreateUserRequest as CreateUserArgs;
+use OpenAPI\Client\Model\WebAuthnDevices;
+use OpenAPI\Client\ApiException;
+
+class User
+{
+    private readonly UsersApi $usersApi;
+    private readonly UserDevicesApi $userDevicesApi;
+    private readonly TokensApi $tokensApi;
+
+    /**
+     * User class for handling operations to get and update user information.
+     */
+    public function __construct(private string $appId, Configuration $config)
+    {
+        $this->appId = $appId;
+        $this->usersApi = new UsersApi(null, $config);
+        $this->userDevicesApi = new UserDevicesApi(null, $config);
+        $this->tokensApi = new TokensApi(null, $config);
+    }
+
+    /**
+     * Get a user's object using their user ID.
+     *
+     * @param string $userId The Passage user ID
+     * @return PassageUser Passage User object
+     * @throws PassageError
+     */
+    public function get(string $userId): PassageUser
+    {
+        try {
+            return $this->usersApi->getUser($this->appId, $userId)->getUser();
+        } catch (ApiException $e) {
+            throw PassageError::fromApiException($e);
+        }
+    }
+
+    /**
+     * Get a user's object using their user identifier.
+     *
+     * @param string $identifier The Passage user email or phone number
+     * @return PassageUser Passage User object
+     * @throws PassageError
+     */
+    public function getByIdentifier(string $identifier): PassageUser
+    {
+        try {
+            $users = $this->usersApi->listPaginatedUsers(
+                $this->appId,
+                limit: 1,
+                identifier: strtolower($identifier)
+            )->getUsers();
+
+            if (count($users) === 0) {
+                throw PassageError::fromApiException(
+                    new ApiException(
+                        null,
+                        404,
+                        null,
+                        "{\"code\":\"user_not_found\",\"error\":\"Could not find user with that identifier.\"}",
+                    ),
+                );
+            }
+
+            $userId = $users[0]->getId();
+            return $this->usersApi->getUser($this->appId, $userId)->getUser();
+        } catch (ApiException $e) {
+            throw PassageError::fromApiException($e);
+        }
+    }
+
+    /**
+     * Activate a user using their user ID.
+     *
+     * @param string $userId The Passage user ID
+     * @return PassageUser Passage User object
+     * @throws PassageError
+     */
+    public function activate(string $userId): PassageUser
+    {
+        try {
+            return $this->usersApi->activateUser($this->appId, $userId)->getUser();
```

Comment:
& looks like the same might be happening here when calling `getUser`:
```
    - Snapshot  - 4
    + Received  + 1

    - [400] Client error: `PATCH https://api.passage.id/v1/apps/***/users//activate` resulted in a `400 Bad Request` response:
    - Invalid format for parameter user_id: parameter 'user_id' is empty, can't bind its value
    -
    -
    + "Warning: Trying to access array offset on null"
```

Reply:
same as https://github.com/passageidentity/passage-php/pull/86#discussion_r1873755293